dotCMS/src/main/java/com/dotcms/content/model/IVanityUrl.java

```diff
@@ -0,0 +1,129 @@
+package com.dotcms.content.model;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.Permissionable;
+import com.dotmarketing.business.Ruleable;
+import com.dotmarketing.business.Treeable;
+import com.dotmarketing.business.Versionable;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+
+
+/**
+ * Represents an Vanity URL in dotCMS. As of version 4.2, Vanity URL are not 
+ * considered as {@link VirtualLink} objects, but as {@link Contentlet} objects.
+ * @author oswaldogallango
+ *
+ */
+public interface IVanityUrl extends Serializable, Versionable, Permissionable, Treeable, Ruleable {
```

Comment:
guys two things here: I think we should start dropping the "I" prefix for interfaces and use type names. Think of "List" which is an interface, it is not "IList". I would have the interface named "VanityUrl" and the one concrete impl we have named "DefaultVanityUrl" for example. And second, for it to make sense having an interface, we need to use it in the VanityUrlAPI, which we are *not*.